File: 2553/2553.cs
public class Solution {
    public int SeparateTheDigitsInAnArray(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(SeparateTheDigitsInAnArray(root.left), SeparateTheDigitsInAnArray(root.right));
    }
}

File: 2553/2553.go
func separateTheDigitsInAnArray(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := separateTheDigitsInAnArray(root.Left)
	r := separateTheDigitsInAnArray(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}